pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/AbstractHierarchicalLedgerManager.java

```diff
@@ -68,7 +68,8 @@ void asyncProcessLevelNodes(
             final AsyncCallback.VoidCallback finalCb, final Object context,
             final int successRc, final int failureRc) {
 
-        store.getChildren(path)
+        store.sync(path)
+                .thenCompose(__ -> store.getChildren(path))
```

Comment:
I don't think that this solves the issue. The reason is that `store.getChildren` will use the cache. The cache is updated asynchronously based on notifications so there's still a chance that the value isn't consistent.

Reply:
Correct. We should be using `store.getChildrenFromStore()` (though it's currently protected), to bypass the cache completely.

Reply:
Updated, could you help review again ? @merlimat @lhotari